junction/proposals/templatetags/proposal_tags.py

```diff
@@ -0,0 +1,16 @@
+from django import template
+register = template.Library()
+from junction.proposals import models
+
+
+@register.assignment_tag
+def get_comment_vote_value(comment_id, user):
+
+    comment_note_value = 0
+    vote = models.ProposalCommentVote.objects.filter(proposal_comment_id=comment_id, voter=user)
+    if vote.exists():
+        if vote[0].up_vote is True:
+            comment_note_value = 1
+        else:
+            comment_note_value = -1
+    return comment_note_value
```

Comment:
Done that you need to invalidate the the vote value for a user, if he/she changes here mind and downvoted a comment. 

You if would be if you abstract the voting and downvoting logic to a function say inside `proposal.services`

```
def add_vote_for_comment(user, comment, vote_value):
  cache_key = "cvote:{user_id}:{comment_id}".format(user_id=user.id, comment_id=comment.id)

  # ... save to db here

  # add/update the cache
  cache.set(cache_key, vote_value) 
```